docs/wip/2026-04-10-intent-drift-detection-design.md

```
@@ -0,0 +1,318 @@
+  async check(params: DriftCheckParams): Promise<DriftVerdict | null>
+
+  /** Returns true if the verdict meets the configured confidence threshold for pausing. */
+  shouldPause(verdict: DriftVerdict): boolean
+}
+```
+
+The `check()` method builds a system + user prompt, calls the LLM with `max_tokens: 200` and `temperature: 0`, and parses the JSON response. It treats any parse failure as a non-drift verdict (fail-open) and logs a warning.
+
+The judge prompt structure:
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)